index.js

```diff
@@ -325,6 +326,13 @@ Primus.readable('initialise', function initialise(Transformer, options) {
   this.use('authorization', require('./middleware/authorization'));
 
   //
+  // Set the heartbeat interval.
+  //
+  if (options.timeout) {
+    this.interval = setInterval(this.heartbeat.bind(this), options.timeout);
```

Comment:
I don't see that as too big of a deal; execution will be very short. So long as it always gets cleaned up properly (and doesn't keep the server open) I don't foresee any ill effects.